groovy-language-server/src/main/java/com/palantir/ls/groovy/GroovycWrapper.java

```diff
@@ -184,17 +255,36 @@ private Pattern getQueryPattern(String query) {
         try {
             return Pattern.compile(newQuery);
         } catch (PatternSyntaxException e) {
-            log.warn("Could not create valid pattern from query {}", query);
+            log.warn("Could not create valid pattern from query '{}'", query);
         }
         return Pattern.compile("^" + escaped);
     }
 
     private void addAllSourcesToCompilationUnit() {
+        // We don't include the files that have a corresponding SourceWriter since that means they will be replaced.
         for (File file : Files.fileTreeTraverser().preOrderTraversal(workspaceRoot.toFile())) {
-            if (file.isFile() && Files.getFileExtension(file.getAbsolutePath()).equals(GROOVY_EXTENSION)) {
+            if (!originalSourceToChangedSource.containsKey(file.toPath()) && file.isFile()
+                    && Files.getFileExtension(file.getAbsolutePath()).equals(GROOVY_EXTENSION)) {
                 unit.addSource(file);
             }
         }
+        // Add the replaced sources
+        originalSourceToChangedSource.values()
+                .forEach(sourceWriter -> unit.addSource(sourceWriter.getDestination().toFile()));
+    }
+
+    private void resetCompilationUnit() {
+        try {
+            FileUtils.deleteDirectory(config.getTargetDirectory());
+            if (!config.getTargetDirectory().mkdir()) {
+                throw new RuntimeException(String.format("Could not recreate target directory: %s",
```

Comment:
hmm we are not allowing users to configure the target directory. Feel it's leaking implementation details